Detectors/TOF/simulation/src/Strip.cxx

```diff
@@ -0,0 +1,107 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+//
+//  Strip.cxx: structure to store the TOF hits and digits in strips - useful
+// for clusterization purposes
+//  ALICEO2
+//
+#include <cstring>
+#include <tuple>
+
+#include <TMath.h>
+#include <TObjArray.h>
+
+#include "TOFSimulation/Strip.h"
+
+using namespace o2::tof;
+
+ClassImp(o2::tof::Strip);
+
+//_______________________________________________________________________
+Strip::Strip(Int_t index)
+  : mStripIndex(index)
+{
+}
+//_______________________________________________________________________
+Strip::Strip(const Strip& ref) = default;
+
+//_______________________________________________________________________
+void Strip::insertHit(const HitType* h)
+{
+  int stripIndex = getStripIndex(h);
+  if (stripIndex != mStripIndex) {
+    LOG(ERROR) << "Strip Index of of requested hit (" << stripIndex << ") does not match with current strip (" << stripIndex << ")"
+               << "\n";
+    return;
+  }
+  mHits.push_back(h);
+}
+
+//_______________________________________________________________________
+const HitType* Strip::getHitAt(Int_t i) const
+{
+  if (i < mHits.size()) {
+    return mHits[i];
+  }
+  return nullptr;
+}
```

Comment:
What is the added value of this function? Consider just using `mHits[]` or `mHits.at`. Such functions should be inlined.

Reply:
Agreed. But it depends where such function is used. If hits are accessed in a tight loop, such approach can be very costly. Here I would not see a problem of returning the `std::vector<>` container to be used directly (or better yet iterators that allow a caller to iterate over the hits ... or a view).

As I said, we can also keep this function but just move it to the header (and using existing mHits.at[]) which even does the bounds checking (should this be wanted). 

Such functions catch my eye because in the past these kind of things showed up significantly in profiles of AliRoot.

Reply:
ok, I see the point. But then it brings the question whether an API function to get a specific hit is needed / advisable at all (e.g. if we always loop on bunch of hits, providing a method for a single hit is just calling for performance issues ?). That is of course not specific to this PR / detector, just food for thought...

Reply:
Now it is inline in the header file, using the "at" function of std vector.